src/common/include/gudhi/graph_simplicial_complex.h

```diff
@@ -30,40 +31,127 @@ struct vertex_filtration_t {
+  Filtered_edge_set edges() const { return edges_; }
+
+  /** \brief Returns the edges vector at a specific index filtration value.
+   */
+  typename SimplicialComplexForFilteredEdges::Filtration_value get_filtration_at(std::size_t new_index) const {
+    return std::get<0>(*(edges_.begin() + new_index));
+  }
+
+  /** \brief Returns the edges vector min filtration value.
+   */
+  typename SimplicialComplexForFilteredEdges::Filtration_value get_filtration_min() const {
+    return std::get<0>(*(edges_.begin()));
+  }
+
+  /** \brief Returns the edges vector max filtration value.
+   */
+  typename SimplicialComplexForFilteredEdges::Filtration_value get_filtration_max() const {
+    return std::get<0>(*(edges_.rbegin()));
+  }
+
+  /** \brief Returns the edges vector.
+   */
+  std::size_t size() const { return edges_.size(); }
+
+  /** \brief Returns the sub-filtered edges vector from a `new_threshold` filtration value.
+   */
+  Filtered_edge_set sub_filter_edges_by_filtration(typename SimplicialComplexForFilteredEdges::Filtration_value new_threshold) const {
+    auto edge_it = edges_.begin();
+    while ((std::get<0>(*edge_it) < new_threshold) && (edge_it < edges_.end())) {
+      ++edge_it;
+    }
+#ifdef DEBUG_TRACES
+    if (edges_.begin() != edge_it) {
+      Filtered_edge_set output(edges_.begin(), edge_it);
+      Filtered_edge back = output.back();
+      std::cout << "Filtered_edges_vector::sub_filter_edges_by_filtration threshold = "
+                << std::get<0>(back) << " - size = " << output.size() << " - back = "
+                << std::get<0>(back) << " - u = " << std::get<1>(back)
+                << " - v = " << std::get<2>(back) << std::endl;
+    } else {
+      std::cout << "Filtered_edges_vector::sub_filter_edges_by_filtration is empty " << std::endl;
+    }
+#endif  // DEBUG_TRACES
+    return Filtered_edge_set(edges_.begin(), edge_it);
+  }
+
+  /** \brief Returns the sub-filtered edges vector from a `new_index` index value.
+   */
+  Filtered_edge_set sub_filter_edges_by_index(std::size_t new_index) const {
+    if (new_index >= size())
+      return edges_;
+
+#ifdef DEBUG_TRACES
+    if (edges_.begin() != edges_.begin() + new_index + 1) {
+      Filtered_edge_set output(edges_.begin(), edges_.begin() + new_index + 1);
+      Filtered_edge back(output.back());
+      std::cout << "Filtered_edges_vector::sub_filter_edges_by_filtration threshold = "
+                << std::get<0>(back) << " - size = " << output.size() << " - back = "
+                << std::get<0>(back) << " - u = " << std::get<1>(back)
+                << " - v = " << std::get<2>(back) << std::endl;
+    } else {
+      std::cout << "Filtered_edges_vector::sub_filter_edges_by_index is empty " << std::endl;
+    }
+#endif  // DEBUG_TRACES
+    return Filtered_edge_set(edges_.begin(), edges_.begin() + new_index + 1);
+  }
+
+private:
+  Filtered_edge_set edges_;
+};
+
+/** \brief Computes the edge graph of the points.
```

Comment:
I am not a fan of the name "edge graph". Having edges is pretty much the definition of a graph... *Rips graph*? *Ball graph*? That is, if you don't like the original name: *proximity graph*.